### PR TITLE
Add merge

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

File: vespajlib/src/main/java/com/yahoo/tensor/functions/Merge.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
```
Comment (lesters) [inline vespajlib/src/main/java/com/yahoo/tensor/functions/Merge.java]: 2019 (2020?) Oath Inc.

File: vespajlib/src/main/java/com/yahoo/tensor/TensorType.java [hunk 2]
```diff
@@ -329,7 +330,7 @@ Dimension combineWith(Optional<Dimension> other) {
             // both are indexed bound
             IndexedBoundDimension thisIb = (IndexedBoundDimension)this;
             IndexedBoundDimension otherIb = (IndexedBoundDimension)other.get();
-            return thisIb.size().get() < otherIb.size().get() ? thisIb : otherIb;
+                return thisIb.size().get() < otherIb.size().get() ? thisIb : otherIb;
```
Comment (lesters) [inline vespajlib/src/main/java/com/yahoo/tensor/TensorType.java]: Extra tab

**Review by lesters — APPROVED** [review]
LGTM. Consider comments.

File: vespajlib/src/main/java/com/yahoo/tensor/functions/Merge.java (hunk 1 above)
Comment (bratseth) [reply vespajlib/src/main/java/com/yahoo/tensor/functions/Merge.java]: Fixed, thanks.

File: vespajlib/src/main/java/com/yahoo/tensor/TensorType.java (hunk 2 above)
Comment (bratseth) [reply vespajlib/src/main/java/com/yahoo/tensor/TensorType.java]: Fixed, thanks.

**Pull request merged**